File: kt/01-02.kt
class Solution {
    fun CheckPermutation(s1: String, s2: String): Boolean {
        val cnt = IntArray(26)
        for (c in s1) {
            cnt[c - 'a']++
        }
        for (c in s2) {
            cnt[c - 'a']--
        }

        return cnt.all { it == 0 }
    }
}

File: rs/01-02.rs
impl Solution {
    pub fn check_permutation(s1: String, s2: String) -> bool {
        let mut cnt = [0; 26];
        for c in s1.bytes() {
            cnt[(c - b'a') as usize] += 1;
        }
        for c in s2.bytes() {
            cnt[(c - b'a') as usize] -= 1;
        }
        
        cnt.iter().all(|&x| x == 0)
    }
}